kernel/rros/smp_test.rs

```diff
@@ -0,0 +1,59 @@
+use crate::{flags::RrosFlag, thread::KthreadRunner};
+
+use kernel::{c_str, prelude::*};
+
+static mut FLAG_PTR: *mut RrosFlag = 0 as *mut RrosFlag;
+
+static mut SMP_KTHREAD_RUNNER_1: KthreadRunner = KthreadRunner::new_empty();
+static mut SMP_KTHREAD_RUNNER_2: KthreadRunner = KthreadRunner::new_empty();
+static mut SMP_KTHREAD_RUNNER_3: KthreadRunner = KthreadRunner::new_empty();
+static mut SMP_KTHREAD_RUNNER_4: KthreadRunner = KthreadRunner::new_empty();
+static mut SMP_KTHREAD_RUNNER_5: KthreadRunner = KthreadRunner::new_empty();
+static mut SMP_KTHREAD_RUNNER_6: KthreadRunner = KthreadRunner::new_empty();
+
+fn kthread_fn(id: i32) {
+    unsafe {
+        (*FLAG_PTR).wait_same();
+    }
+
+    let (mut first, mut second) = (1, 1);
+    const M: i64 = 1_000_000_007;
+    for _ in 0..5_000_000_0 {
+        let temp = (first + second) % M;
+        first = second;
+        second = temp;
+    }
+    pr_warn!(
+        "[smp_test]: kthread id is {:?}, the calculation result is {:?}",
+        id,
+        second
+    );
+}
+
+pub fn test_smp() {
```

Comment:
Add a `dead_code` attribute. Move this to the init.rs just like other tests.

Reply:
Ok, I will move this function to `init.rs`. But I don't understand why to add a `dead_code` attribute. Do you mean this test function shouldn't be called in the kernel `init()` function? If so, does the `oobcpus_arg` parameter need to be changed back to `"0"`?

Reply:
Because we may comment these test functions. Adding a test macro is also fine.

Reply:
OK, I get it.